### PR TITLE
Create Total TLS Resource

### LINKED ISSUE #1967: Total TLS support
### Current Terraform and Cloudflare provider version

Terraform v1.3.2
on darwin_arm64

### Description

Cloudflare releases a new feature for Total TLS.

Cloudflare already provides the API to view/change it: [https://api.cloudflare.com/#total-tls-enable-or-disable-total-tls](https://api.cloudflare.com/#total-tls-enable-or-disable-total-tls)

It would be awesome to be able to configure this via terraform when managing multiple zones.

It can likely just be a new key/value here: [https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/zone_settings_override](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/zone_settings_override)

### Use cases

Enable/Disable Total TLS on a zone

### Potential Terraform configuration

```hcl
resource "cloudflare_zone_settings_override" "test" {
    settings {
        total_tls = "google" # or "lets_encrypt" or "off"
    }
}
```

### References

Announcement: [https://blog.cloudflare.com/total-tls-one-click-tls-for-every-hostname/](https://blog.cloudflare.com/total-tls-one-click-tls-for-every-hostname/)

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
Creates a new resource to manage the Total TLS setting. 
Resolves #1967

### CODE REVIEW

**Comment by Cyb3r-Jak3** [comment]
This does work when replacing cloudflare-go with latest master
```
root@374a6841e75f:/workspaces/terraform-provider-cloudflare# TESTARGS='-run "^TestAccCloudflareTotalTLS" -count 1 -parallel 1' make testacc
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTotalTLS" -count 1 -parallel 1 -timeout 120m -parallel 1
?       github.com/cloudflare/terraform-provider-cloudflare     [no test files]
=== RUN   TestAccCloudflareTotalTLS
--- PASS: TestAccCloudflareTotalTLS (2.84s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/provider   2.856s
```

File: docs/resources/total_tls.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+page_title: "cloudflare_total_tls Resource - Cloudflare"
+subcategory: ""
+description: |-
+  
+---
+
+# cloudflare_total_tls (Resource)
+
+
```
Comment (jacobbednarz) [inline docs/resources/total_tls.md:10]: looks like we're missing a schema description here.

File: internal/provider/resource_cloudflare_total_tls.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareTotalTLS() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareTotalTLSSchema(),
+		CreateContext: resourceCloudflareTotalSSLUpdate,
+		ReadContext:   resourceCloudflareTotalSSLRead,
+		UpdateContext: resourceCloudflareTotalSSLUpdate,
+		DeleteContext: resourceCloudflareTotalSSLDelete,
```
Comment (jacobbednarz) [inline internal/provider/resource_cloudflare_total_tls.go:17]: we'll want to add a description here.

Comment (vcajes) [reply internal/provider/resource_cloudflare_total_tls.go:19]: @jacobbednarz it seems the description was added :)

**Comment by jacobbednarz** [comment]
acceptance tests looking good

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTotalTLS" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareTotalTLS
--- PASS: TestAccCloudflareTotalTLS (10.33s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/provider	10.920s
```

**Comment by jacobbednarz** [comment]
appreciate the effort here @Cyb3r-Jak3 🏆

**Pull request merged**